src/kpro_lib.erl

```diff
@@ -226,19 +223,35 @@ get_prelude_schema(Tag, Vsn) ->
 -spec find(kpro:field_name(), kpro:struct()) -> kpro:field_value().
 find(Field, Struct) when is_map(Struct) ->
   try
-    maps:get(Field, Struct)
+      try 
+          maps:get(Field, Struct)
+      catch
+          error : {badkey, _} when Field =:= auth_bytes ->
+              maps:get(sasl_auth_bytes, Struct);
+          error : {badkey, _} when Field =:= sasl_auth_bytes ->
+              maps:get(auth_bytes, Struct)
+      end
```

Comment:
maybe this:
find2(Field, Struct) ->
    try find(Field, Struct)
    catch error : {no_such_field, auth_bytes} ->
        find(sasl_auth_bytes, Struct)
    end.
```

Reply:
Ok, I will try to find the places where it is needed and only use find2 there.

Reply:
Hmm, not sure that this is the right thing anymore. The find function function is used by the brod_gssapi plugin https://github.com/kafka4beam/brod_gssapi/blob/master/src/brod_gssapi_v1.erl#L161 which complicates things. Do you still think it is a good idea to introduce a new function. What are the reasons for that in that case? I like your comment and think it will be enough to add it to the appropriate place in the `find` function.

Reply:
Thanks fixed, see [245aae4](https://github.com/emqx/kafka_protocol/pull/2/commits/245aae40424c19b75bc42a29b45d61a7d9433565)